### PR TITLE
Fix block movers in navigator on experimental navigation page displaying horizontally

### PR DESCRIPTION
**Pull request by talldan**
## Description
Fixes two small issues from https://github.com/WordPress/gutenberg/pull/23416.

1. `BlockMobileToolbar` (which is the web-based responsive mobile toolbar) had a `moverDirection` prop, which wasn't being used and didn't do anything, as `BlockMover` handles its own orientation so this can be omitted.
2. `BlockMoverUpButton` and `BlockMoverDownButton` are both used in the block navigator on the experimental navigation page. The prop for these components should have been renamed to `orientation` from `__experimentalOrientation` but was somehow missed. This fixes an issue where the block movers in the navigator are sometimes shown horizontally instead of vertically—the prop is used to force the movers to display vertically.

## How has this been tested?
1. Open up the experimental navigation page and create a menu with a few links
2. Try the movers in the Navigation Structure area.
3. Observe the movers are correctly oriented.

#### Before
Screenshot 2020-07-08 at 12 53 30 pm

#### After
Screenshot 2020-07-08 at 12 57 32 pm

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/block-mobile-toolbar/index.js [hunk 1]
```diff
@@ -8,18 +8,15 @@ import { useViewportMatch } from '@wordpress/compose';
  */
 import BlockMover from '../block-mover';
 
-function BlockMobileToolbar( { clientId, moverDirection } ) {
+function BlockMobileToolbar( { clientId } ) {
```
Comment (talldan) [inline packages/block-editor/src/components/block-mobile-toolbar/index.js:11]: I think this file can actually be deleted. I couldn't see any usage other than the native mobile version of it.

Comment (noisysocks) [reply packages/block-editor/src/components/block-mobile-toolbar/index.js:11]: Wouldn't deleting it break the native mobile app, then?

Comment (talldan) [reply packages/block-editor/src/components/block-mobile-toolbar/index.js:11]: Yeah, that's what I don't know. I don't think so as there's a separate index.native.js.

Comment (noisysocks) [reply packages/block-editor/src/components/block-mobile-toolbar/index.js:11]: Aha, gotcha. Yeah you're right that it probably can be removed.

**Review by noisysocks — APPROVED** [review]
Looks good!

**Pull request merged**